diff --git policycoreutils-2.9/po/Makefile policycoreutils-2.9/po/Makefile
index 575e143..18bc1df 100644
--- policycoreutils-2.9/po/Makefile
+++ policycoreutils-2.9/po/Makefile
@@ -3,7 +3,6 @@
 #
 
 PREFIX ?= /usr
-TOP	 = ../..
 
 # What is this package?
 NLSPACKAGE	= policycoreutils
@@ -32,74 +31,13 @@ USE_LINGUAS := $(if $(USER_LINGUAS),$(USER_LINGUAS),$(PO_LINGUAS))
 
 POFILES		= $(patsubst %,%.po,$(USE_LINGUAS))
 MOFILES		= $(patsubst %.po,%.mo,$(POFILES))
-POTFILES = \
-	../run_init/open_init_pty.c \
-	../run_init/run_init.c \
-	../semodule_link/semodule_link.c \
-	../audit2allow/audit2allow \
-	../semanage/seobject.py \
-	../setsebool/setsebool.c \
-	../newrole/newrole.c \
-	../load_policy/load_policy.c \
-	../sestatus/sestatus.c \
-	../semodule/semodule.c \
-	../setfiles/setfiles.c \
-	../semodule_package/semodule_package.c \
-	../semodule_deps/semodule_deps.c \
-	../semodule_expand/semodule_expand.c \
-	../scripts/chcat \
-	../scripts/fixfiles \
-	../restorecond/stringslist.c \
-	../restorecond/restorecond.h \
-	../restorecond/utmpwatcher.h \
-	../restorecond/stringslist.h \
-	../restorecond/restorecond.c \
-	../restorecond/utmpwatcher.c \
-	../gui/booleansPage.py \
-	../gui/fcontextPage.py \
-	../gui/loginsPage.py \
-	../gui/mappingsPage.py \
-	../gui/modulesPage.py \
-	../gui/polgen.glade \
-	../gui/polgengui.py \
-	../gui/portsPage.py \
-	../gui/semanagePage.py \
-	../gui/statusPage.py \
-	../gui/system-config-selinux.glade \
-	../gui/system-config-selinux.py \
-	../gui/usersPage.py \
-	../secon/secon.c \
-	booleans.py \
-	../sepolicy/sepolicy.py \
-	../sepolicy/sepolicy/communicate.py \
-	../sepolicy/sepolicy/__init__.py \
-	../sepolicy/sepolicy/network.py \
-	../sepolicy/sepolicy/generate.py \
-	../sepolicy/sepolicy/sepolicy.glade \
-	../sepolicy/sepolicy/gui.py \
-	../sepolicy/sepolicy/manpage.py \
-	../sepolicy/sepolicy/transition.py \
-	../sepolicy/sepolicy/templates/executable.py \
-	../sepolicy/sepolicy/templates/__init__.py \
-	../sepolicy/sepolicy/templates/network.py \
-	../sepolicy/sepolicy/templates/rw.py \
-	../sepolicy/sepolicy/templates/script.py \
-	../sepolicy/sepolicy/templates/semodule.py \
-	../sepolicy/sepolicy/templates/tmp.py \
-	../sepolicy/sepolicy/templates/user.py \
-	../sepolicy/sepolicy/templates/var_lib.py \
-	../sepolicy/sepolicy/templates/var_log.py \
-	../sepolicy/sepolicy/templates/var_run.py \
-	../sepolicy/sepolicy/templates/var_spool.py
+POTFILES  = $(shell cat POTFILES)
 
 #default:: clean
 
-all::  $(MOFILES)
+all:: $(POTFILE) $(MOFILES)
 
-booleans.py:
-	sepolicy booleans -a > booleans.py
-
-$(POTFILE): $(POTFILES) booleans.py
+$(POTFILE): $(POTFILES)
 	$(XGETTEXT) --keyword=_ --keyword=N_ $(POTFILES)
 	@if cmp -s $(NLSPACKAGE).po $(POTFILE); then \
 	    rm -f $(NLSPACKAGE).po; \
@@ -107,8 +45,6 @@ $(POTFILE): $(POTFILES) booleans.py
 	    mv -f $(NLSPACKAGE).po $(POTFILE); \
 	fi; \
 
-update-po: Makefile $(POTFILE) refresh-po
-	@rm -f booleans.py
 
 refresh-po: Makefile
 	for cat in $(POFILES); do \
diff --git policycoreutils-2.9/po/POTFILES policycoreutils-2.9/po/POTFILES
new file mode 100644
index 0000000..12237dc
--- /dev/null
+++ policycoreutils-2.9/po/POTFILES
@@ -0,0 +1,9 @@
+../run_init/open_init_pty.c
+../run_init/run_init.c
+../setsebool/setsebool.c
+../newrole/newrole.c
+../load_policy/load_policy.c
+../sestatus/sestatus.c
+../semodule/semodule.c
+../setfiles/setfiles.c
+../secon/secon.c
diff --git policycoreutils-2.9/scripts/fixfiles policycoreutils-2.9/scripts/fixfiles
index b277958..53d28c7 100755
--- policycoreutils-2.9/scripts/fixfiles
+++ policycoreutils-2.9/scripts/fixfiles
@@ -108,6 +108,7 @@ exclude_dirs_from_relabelling() {
 fullFlag=0
 BOOTTIME=""
 VERBOSE="-p"
+[ -t 1 ] || VERBOSE=""
 FORCEFLAG=""
 RPMFILES=""
 PREFC=""
diff --git policycoreutils-2.9/setfiles/setfiles.8 policycoreutils-2.9/setfiles/setfiles.8
index ccaaf4d..a8a76c8 100644
--- policycoreutils-2.9/setfiles/setfiles.8
+++ policycoreutils-2.9/setfiles/setfiles.8
@@ -57,7 +57,7 @@ check the validity of the contexts against the specified binary policy.
 .TP
 .B \-d
 show what specification matched each file (do not abort validation
-after ABORT_ON_ERRORS errors).
+after ABORT_ON_ERRORS errors). Not affected by "\-q"
 .TP
 .BI \-e \ directory
 directory to exclude (repeat option for more than one directory).
