diff --exclude-from=exclude -N -u -r nsasepolgen/src/sepolgen/access.py policycoreutils-2.0.60/sepolgen-1.0.14/src/sepolgen/access.py
--- nsasepolgen/src/sepolgen/access.py  2008-08-28 09:34:24.000000000 -0400
+++ policycoreutils-2.0.60/sepolgen-1.0.14/src/sepolgen/access.py       2008-12-01 11:41:09.000000000 -0500
@@ -314,7 +314,7 @@

     def __len__(self):
         """Return the unique number of role allow statements."""
-        return len(self.roles)
+        return len(self.role_type.keys())

     def add(self, role, type):
         if self.role_types.has_key(role):
diff --exclude-from=exclude -N -u -r nsasepolgen/src/sepolgen/refparser.py policycoreutils-2.0.60/sepolgen-1.0.14/src/sepolgen/refparser.py
--- nsasepolgen/src/sepolgen/refparser.py       2008-08-28 09:34:24.000000000 -0400
+++ policycoreutils-2.0.60/sepolgen-1.0.14/src/sepolgen/refparser.py    2008-12-01 11:26:20.000000000 -0500
@@ -919,7 +919,7 @@
 def list_headers(root):
     modules = []
     support_macros = None
-    blacklist = ["init.if", "inetd.if", "uml.if", "thunderbird.if"]
+    blacklist = ["uml.if", "thunderbird.if"]

     for dirpath, dirnames, filenames in os.walk(root):
         for name in filenames:
