diff --git semodule-utils-2.9/semodule_package/semodule_package.c semodule-utils-2.9/semodule_package/semodule_package.c
index 3515234..7b75b3f 100644
--- semodule-utils-2.9/semodule_package/semodule_package.c
+++ semodule-utils-2.9/semodule_package/semodule_package.c
@@ -74,6 +74,7 @@ static int file_to_data(const char *path, char **data, size_t * len)
 	}
 	if (!sb.st_size) {
 		*len = 0;
+		close(fd);
 		return 0;
 	}
 
